TGC.MonoGame.Samples/Samples/Collisions/CameraCollision.cs

```diff
@@ -0,0 +1,380 @@
+            var normalizedDifference = difference / distanceToRobot;
+
+            Ray cameraToPlayerRay = new Ray(cameraPosition, normalizedDifference);
+
+            // Test our ray against every wall Bounding Box
+            for (int index = 0; index < WallBoxes.Length; index++)
+            {
+                // If there was an intersection 
+                // And this intersection happened between the Robot and the Camera (and not further away)
+                // Return the distance of collision
+                var distance = cameraToPlayerRay.Intersects(WallBoxes[index]);
+                if (distance.HasValue && distance < distanceToRobot)
+                    return distance;
+            }
+            // Return null if there was no valid collision
+            return null;
+        }
+
+        /// <inheritdoc />
+        public override void Update(GameTime gameTime)
+        {
+            // Initialize values indicating if the Robot moved, if it rotated, and how much movement should be applied
+            var advanceAmount = 0f;
+            var rotated = false;
+            var moved = false;
+
+            // Check for key presses and rotate accordingly
+            // We can stack rotations in a given axis by multiplying our past matrix
+            // By a new matrix containing a new rotation to apply
+            if (Game.CurrentKeyboardState.IsKeyDown(Keys.Right))
+            {
+                RobotRotation *= Matrix.CreateRotationY(-RobotRotatingVelocity);
+                rotated = true;
+            }
+            else if (Game.CurrentKeyboardState.IsKeyDown(Keys.Left))
+            {
+                RobotRotation *= Matrix.CreateRotationY(RobotRotatingVelocity);
+                rotated = true;
+            }
+
+
+            // Check for key presses and set our advance variable accordingly
+            if (Game.CurrentKeyboardState.IsKeyDown(Keys.Up))
+            {
+                advanceAmount = RobotVelocity;
+                moved = true;
+            }
+            else if (Game.CurrentKeyboardState.IsKeyDown(Keys.Down))
+            {
+                advanceAmount = -RobotVelocity;
+                moved = true;
+            }
+
+            // If there was any movement
+            if (moved)
+            {
+                // Calculate the Robot new Position using the last Position, 
+                // And adding an increment in the Robot Direction (calculated by rotating a vector by its rotation)
+                var newPosition = RobotPosition + Vector3.Transform(Vector3.Backward, RobotRotation) * advanceAmount;
+
+                // Set the Center of the Cylinder as our calculated position
+                RobotCylinder.Center = newPosition;
+
+                // Test against every wall. If there was a collision, move our Cylinder back to its original position
+                for (int index = 0; index < WallBoxes.Length; index++)
```

Comment:
cambniarlo por foreach (var box in WallBoxes)?

Reply:
Ya discutimos que no esto, creo.